packages/apollo-gateway/src/index.ts

```diff
@@ -239,37 +245,50 @@ export class ApolloGateway implements GraphQLService {
     };
   }
 
-  protected async updateComposition(options?: {
-    engine?: GraphQLServiceEngineConfig;
-  }) {
+  private async updateComposition(): Promise<void> {
     const previousSchema = this.schema;
     const previousServiceDefinitions = this.serviceDefinitions;
     const previousCompositionMetadata = this.compositionMetadata;
 
-    if (options && options.engine) {
-      if (!options.engine.graphVariant)
-        console.warn('No graph variant provided. Defaulting to `current`.');
-      this.engineConfig = options.engine;
+    let result: Await<ReturnType<Experimental_UpdateServiceDefinitions>>;
+    try {
+      this.logger.debug('Loading configuration for gateway');
```

Comment:
Limiting the concern of the `try` to precisely what it's trying to `catch` is, I think, a good practice.  In other words, any reason to not put this log message before the `try`?

Reply:
Agree! Resolved in 9c2b701